central/networkgraph/aggregator/aggregator_impl.go

```diff
@@ -316,3 +320,14 @@ func normalizeDupNameExtSrcs(entity *storage.NetworkEntityInfo) {
 		},
 	}
 }
+
+// Replace the pointed entity with NetworkEntityInfo_INTERNET if:
+// - it is a 'discovered' entity
+// - displaying of 'discovered' entities is disabled
+func anonymizeIfNoExternalIPs(entity *storage.NetworkEntityInfo) {
+	if networkgraph.IsExternalDiscovered(entity) && !features.NetworkGraphExternalIPs.Enabled() {
+		// If this is a discovered entity, but network graph display is disabled
+		// we aggregate into the internet entity
+		*entity = *networkgraph.InternetEntity().ToProto()
+	}
+}
```

Comment:
I want to summarize our discussion from Today. I tried to understand for over 2 hours what the side effects of the anonymization are doing, but I failed. I would like to see another version of this function that:

1. Does not rely on the feature flag - it should be passed as a param.
2. Does not rely on `*lvalue = sth`
3. Returns an `entity *storage.NetworkEntityInfo`
4. The object returned from that func is not used in `*lvalue = sth`

This is for me a blocking issue, as it turns out, I do not understand what the anonymization does. I tried hard to write a unit test for the act of anonymization, but I failed - the test `TestAnonymizeDiscoveredExtEntities` was failing with a proto diff that is very difficult to understand without careful analysis.

I want that we drop all attempts of performance optimization and focus on the clarity of the code. Once we are sure that it does what we think it does, we may look into the performance.

Reply:
This has been followed-up upon, so I am resolving this conversation to unlock merging.